azkaban-common/src/main/java/azkaban/jobExecutor/ProcessJob.java

```diff
@@ -230,6 +240,11 @@ public void run() throws Exception {
             String.format("Not permitted to proxy as '%s' through Azkaban", effectiveUser)
         );
       }
+      // Set parent directory permissions to <uid>:azkaban so user can write in their execution directory
+      // if the directory is not permissioned correctly already (should happen once per execution)
+      if (!canWriteInCurrentWorkingDirectory(effectiveUser)) {
```

Comment:
Wouldn't it be cleaner to create the directory with the proper permission in the first place? I understand you tried and had issues. It would be good to document what the issue is and fix it if possible.